src/py42/services/audit_logs.py

```diff
@@ -0,0 +1,63 @@
+from py42 import settings
+from py42.services import BaseService
+from py42.services.util import get_all_pages
+from py42.util import convert_datetime_to_timestamp_str
+
+
+class AuditLogsService(BaseService):
+    def get_page(self, page_num, page_size=None, **kwargs):
+
+        uri = u"/rpc/search/search-audit-log"
+        page_size = page_size or settings.items_per_page
+        params = dict(
+            page=page_num - 1,
+            pageSize=page_size,
+            dateRange=kwargs["date_range"],
+            eventTypes=kwargs["event_types"],
+            actorIds=kwargs["user_ids"],
+            actorNames=kwargs["user_names"],
+            actorIpAddresses=kwargs["user_ip_addresses"],
+            affectedUserIds=kwargs["affected_user_ids"],
+            affectedUserNames=kwargs["affected_user_names"],
+        )
+        params["type$"] = "audit_log::audit_log_queries.search_audit_log/1"
+        return self._connection.post(uri, json=params,)
+
+    def get_all(
+        self,
+        begin_time=None,
+        end_time=None,
+        event_types=None,
+        user_ids=None,
+        user_names=None,
+        user_ip_addresses=None,
+        affected_user_ids=None,
+        affected_user_names=None,
+    ):
+        comma = u","
```

Comment:
I think it makes more sense for all of these to be arrays instead of strings that got split on a comma. We can also accept plain strings and turn them into arrays of just one item.

Reply:
Done.